drivers/modem/murata-1sc.c

```diff
@@ -0,0 +1,4499 @@
+/*
+ * Copyright (c) 2023 T-Mobile USA, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT murata_1sc
+#define MURATA_1SC_C
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(modem_murata_1sc, CONFIG_MODEM_LOG_LEVEL);
+
+#include "modem_cmd_handler.h"
+#include "modem_context.h"
+#include "modem_iface_uart.h"
+#include "modem_receiver.h"
+#include "modem_socket.h"
+#include "murata-1sc.h"
+
+#include <stdarg.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <zephyr/device.h>
+#include <zephyr/posix/fcntl.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_offload.h>
+#include <zephyr/net/socket_offload.h>
+#include <zephyr/net/offloaded_netdev.h>
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+#include "tls_internal.h"
+#include <zephyr/net/tls_credentials.h>
+#include <zephyr/sys/base64.h>
+#endif
+#if defined(CONFIG_PM_DEVICE)
+#include <zephyr/pm/device.h>
+#endif
+
+#define MAX_FILENAME_LEN      32
+#define MDM_BOOT_DELAY	      6	   /* seconds */
+#define MDM_WAKE_DELAY	      5000 /* milliseconds */
+#define MDM_CMD_RSP_TIME      K_SECONDS(2)
+#define MDM_CMD_LONG_RSP_TIME K_SECONDS(6)
+
+const struct mdmdata_cmd_t cmd_pool[] = {{"APN", apn_e},
+					 {"ASLEEP", awake_e},
+					 {"AWAKE", awake_e},
+					 {"CONN_STS", connsts_e},
+					 {"CONN", connsts_e},
+					 {"EDRX", edrx_e},
+					 {"GOLD", golden_e},
+					 {"GOLDEN", golden_e},
+					 {"ICCID", iccid_e},
+					 {"IMEI", imei_e},
+					 {"IMSI", imsi_e},
+					 {"IP", ip_e},
+					 {"IP6", ip6_e},
+					 {"MSISDN", msisdn_e},
+					 {"PSM", psm_e},
+					 {"SLEEP", sleep_e},
+					 {"SSI", ssi_e},
+					 {"STAT", connsts_e},
+					 {"SIM", sim_info_e},
+					 {"VERSION", version_e},
+					 {"VER", version_e},
+					 {"WAKE", wake_e},
+					 {}};
+
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+
+#define CERTCMD_WRITE_SIZE 32 + MAX_FILENAME_LEN /* assume filename maxlen = 32 */
+#define PEM_BUFF_SIZE	   3002			 /* terminate with \" & 0 */
+
+/**
+ * following struct may not have packed memory if it has something like
+ * int follow by char then int,
+ * since array of uchar plus array of uchar will be packed
+ */
+struct cert_cmd_t {
+	uint8_t cert_cmd_write[CERTCMD_WRITE_SIZE];
+	uint8_t pem_buf[PEM_BUFF_SIZE];
+};
+
+struct mdm_sock_tls_s {
+	char host[CONFIG_MURATA_MODEM_SNI_BUFFER_SZ + 1];
+	uint8_t profile;
+	bool sni_valid;
+	bool peer_verify_disable;
+	bool client_verify;
+} murata_sock_tls_info[MDM_MAX_SOCKETS] = {{{0}, 0, false, false, false}};
+#endif /* defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS) */
+
+/**
+ * @brief Convert a series of uint8_t or byte to ascii hex value in a string
+ */
+static size_t data_to_hex_str(const void *input_buf, size_t input_len, char *output_buf,
```

Comment:
Not sure if [bin2hex](https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/os/hex.c#L40) can do this for you?